internal/events/handler.go

```diff
@@ -151,3 +156,27 @@ func (h *EventHandlerImpl) propagateDelete(e *DeleteEvent) {
 		panic(fmt.Errorf("unknown resource type %T", e.Type))
 	}
 }
+
+// setProgrammedStatus will set the Gateway and HTTPRoute conditions for when the Gateway is not programmed,
+// due to an error when attempting to reload nginx
+func setProgrammedStatus(objects []client.Object, statuses *state.Statuses) {
```

Comment:
Let me see if I can play around with that. I'd prefer to not be in a situation where we are passing a boolean around to determine if we need to add the error status or not.

Reply:
> I wonder about moving statuses.go into the status package, instead of living in state. What do you think?

> I'm thinking about having the StatusUpdater build the statuses before setting them, rather than building them in the change processor.

To be honest, considering those FIXMEs
https://github.com/nginxinc/nginx-kubernetes-gateway/blob/3d6b009334aaac05c0efcdae187cd7f0993f2b87/internal/status/updater.go#L43C2-L75 , I've been thinking about simplifying StatusUpdater (so that it can reliably update status for any resource -- it doesn't have any resource specific logic) and moving final status generation to the `state` package, so that there is no intermediate status state

Reply:
The thing I'm thinking about is the fact that if we want all statuses to be built at the same time, rather than appending the Programmed: False status after the fact, we need to wait until after we've attempted the nginx reload in order to do so. So building the statuses would no longer be a part of the `ChangeProcessor Process()` function. Since we don't currently save the generated statuses, it doesn't feel that it's needed in the `state` package.

There's likely a lot of work to improve the status updater, per your comments, though that should be its own story, since this one is currently blocking conformance tests.

Reply:
> There's likely a lot of work to improve the status updater, per your comments, though that should be its own story, since this one is currently blocking conformance tests.

kk. that makes sense